--- snvec/snvec-3.7.5.c_bak	2024-12-10 18:06:34.418793238 -1000
+++ snvec/snvec-3.7.5.c	2024-12-10 18:10:59.800158890 -1000
@@ -422,7 +422,7 @@
 {
 int i,k,m,lt;	
 double t0,tfin,tfink,sckx,*nv,*y0,*u,*up,*epl,*phi,tmp,**nni,dx;
-double *eei,*inci,*lphi,*lphu,*lani,*lanu,*cp;
+double *eei,*inci,*lphi,*lphu,*lani,*lanu,*varpi,*omegabar,*cp;
 double argd[4],tend,ed,td;
 char dir[BUFSIZ],foo[BUFSIZ],mssg[BUFSIZ];
 FILE *fpout;
@@ -515,6 +515,8 @@
 lani = dvector(1,kmax);
 lphu = dvector(1,ls);   /* 1,ls! lphu = unwrap(lph) */
 lanu = dvector(1,ls);   /* 1,ls! lanu = unwrap(lan) */
+varpi = dvector(1,kmax); /* longitude of perihelion */
+omegabar = dvector(1,kmax); /* longitude of perihelion with respect to the ascending node */
 cp   = dvector(1,kmax); /* climatic precession      */
 
 /* start values */
@@ -588,7 +590,9 @@
 tmp = phi[1];	
 for(i=1;i<=kount;i++){
   phi[i] -= tmp;
-   cp[i]  = eei[i]*sin((lphi[i]+OMT)/R2D - phi[i]);
+  varpi[i] = (lphi[i]+OMT)/R2D;
+  omegabar[i] = varpi[i] - phi[i];
+   cp[i]  = eei[i]*sin(omegabar[i]);
 }
 
 printf("\n@ Final values obliquity, precession (rad):\n");
@@ -605,10 +609,11 @@
  for(k=1;k<=NEQ;k++)
   fprintf(fpout,frmt,yy[k][i]);
  }
- //fprintf(fpout,frmt,eei[i]);
+ fprintf(fpout,frmt,eei[i]);
  fprintf(fpout,frmt,epl[i]);
  fprintf(fpout,frmt,phi[i]);
  fprintf(fpout,frmt,cp[i]);
+ fprintf(fpout,frmt,omegabar[i]);
  //fprintf(fpout,frmt,lani[i]);
  fprintf(fpout,"\n");
 }
